p2p/host/basic/basic_host.go

```diff
@@ -230,6 +233,7 @@ func New(net network.Network, opts ...interface{}) *BasicHost {
 	}
 
 	h, err := NewHost(context.Background(), net, hostopts)
+	h.Start()
```

Comment:
this seems like a functional change. are we confident this is safe?

Reply:
Hey, thus function is ONLY used in tests.

Reply:
yep. just worth checking where it's currently used to make sure the tests making this downstream aren't either manually starting it as well, or not expecting the changed behavior.

Reply:
@willscott Good point.

So, outside this `go-libp2p` package, it only seems to be used in IPFS for one test(it passes) and in DHT tests(no problem there).

Reply:
Yeah, this should be fine. We should also look into removing it eventually...